code/game/objects/effects/landmarks_events.dm

```diff
@@ -0,0 +1,129 @@
+/*
+Landmark to be spawned by GMs and admins when running events.
+Do NOT directly spawn "" or buildmode spawn this.
+Use the "Manage Event Triggers" verb under "Eventkit" category.
+Admin verb is called by code\modules\admin\verbs\event_triggers.dm
+*/
+/obj/effect/landmark/event_trigger
+	name = "ONLY FOR EVENTS. DO NOT USE WHEN MAPPING"
+	desc = "Please notify staff if you see this!"
+	var/creator_ckey = ""
+	var/isTeamwork = FALSE //Whether to notify creator or whole team.
+	var/isRepeating = FALSE
+	var/coordinates = ""
+	var/cooldown = 0 //in seconds
+	var/last_trigger = 0
+	var/isLoud = FALSE
+	var/isNarrate = FALSE
+
+/obj/effect/landmark/event_trigger/New()
+	..()
+	coordinates = "(X:[loc.x];Y:[loc.y];Z:[loc.z])"
+	invisibility = 100
+
+/obj/effect/landmark/event_trigger/proc/set_vars(mob/M)
+	name = sanitize(tgui_input_text(M, "Input Name for the trigger", "Naming", "Event Trigger"))
```

Comment:
Will cancelling input not result in a blank name here?

Reply:
Good point, I'll make cancelling return FALSE

Reply:
Just have a temporary var for name and check like if(!var) return or something